Userland/Libraries/LibM/complex.h

```diff
@@ -0,0 +1,307 @@
+/*
+ * Copyright (c) 2021, Cesar Torres <shortanemoia@protonmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <AK/Concepts.h>
+#include <math.h>
+
+template<typename T>
+requires AK::IsArithmetic<T>::value class [[gnu::packed]] complex {
```

Comment:
The Serenity coding style says to use CamelCase(first letter is capital) names for classes, i.e. this should be called `Complex`.

Also I think clang-format is formatting the `requires` clause really weird, maybe it'd be worth adding an `Arithmetic` concept to `AK/Concepts.h` so we could write `template<Arithmetic T>` instead? Something like
```cpp
template<typename T>
concept Arithmetic = IsArithmetic<T>::value;
```
would work, I think.

Finally there's no reason to qualify things from AK with the `AK` namespace

Reply:
I put it in lowercase in concordance with other types like the ones under AK::SIMD. I think it's worth making an exception for the complex number type, since having it look like as if it was a native type I think makes reading and comprehending algorithms easier, since you don't have to think that it is another class but just a complex number.
If it can not be avoided then I'll make it camelcase.

Reply:
That makes sense, in that case, I think it's fine.